src/main/java/org/folio/service/titles/TitleInstanceService.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.service.titles;
+
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.Title;
+import org.folio.rest.tools.utils.TenantTool;
+import org.folio.service.inventory.InventoryInstanceManager;
+
+import io.vertx.core.Future;
+
+public class TitleInstanceService {
+
+  private final InventoryInstanceManager inventoryInstanceManager;
+
+  public TitleInstanceService(InventoryInstanceManager inventoryInstanceManager) {
+    this.inventoryInstanceManager = inventoryInstanceManager;
+  }
+
+  public Future<String> getOrCreateInstance(Title title, RequestContext requestContext) {
+    return getOrCreateInstance(title, false, requestContext);
+  }
+
+  public Future<String> getOrCreateInstance(Title title, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
+    return createShadowInstance(title.getInstanceId(), requestContext)
+      .compose(shadowInstId -> createInventoryInstance(shadowInstId, title, isInstanceMatchingDisabled, requestContext));
+  }
```

Comment:
I assume true case can be used in the future, because all places is using false case and increase code chain.

Reply:
@azizbekxm so do I get this right - you are suggesting passing the flag with every call and remove the method with false as default?

Reply:
maybe action not required. I am going to say is that If true isn't used in the future, it would be better to simply methods by removing boolean attributes and just calling directly with false.